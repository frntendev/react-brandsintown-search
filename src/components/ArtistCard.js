import React from "react";
import PropTypes from "prop-types";
import { css } from "react-emotion";

const styles = {
  root: css`
    display: flex;
    position: relative;
    width: 100%;
    height: 400px;
    background: #eeeeee85;
    padding: 10px;
    border-radius: 4px;
    box-sizing: border-box;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
    margin-bottom: 20px;
    transition: 0.3s all ease;
    overflow: hidden;
    label: artistRoot;
    &:hover {
      box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
    }
    @media (max-width: 758px) {
      flex-flow: column;
      height: auto;
    }
  `,
  icon: css`
    width: 20px;
    height: 20px;
    position: absolute;
    right: 15px;
    top: 15px;
    fill: #48c5b6;
    cursor: pointer;
  `,
  artistImage: css`
    min-width: 300px;
    display: flex;
    justify-content: center;
    align-items: baseline;
    border-radius: 4px;
    overflow: hidden;
    background: #dcdcdc;
    label: artistImage;
    @media (max-width: 758px) {
      height: 300px;
    }
    img {
      @media (max-width: 480px) {
        width: 100%;
      }
    }
  `,
  artistDetails: css`
    flex: 1;
    display: flex;
    flex-flow: column;
    justify-content: space-evenly;
    label: artistDetails;
    @media (max-width: 758px) {
      margin-top: 20px;
    }
    span {
      font-size: 1.5rem;
      color: #545454;
      font-style: italic;
      margin-left: 20px;
      @media (max-width: 758px) {
        font-size: 1rem;
        padding: 5px;
      }
    }
  `
};

const ArtistCard = props => {
  return (
    <div id="artist-card" className={styles.root}>
      {props.showClose && (
        <svg
          className={styles.icon}
          viewBox="0 0 16 16"
          onClick={() => props.onClose(props.id)}
        >
          <path d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z" />
        </svg>
      )}
      <div className={styles.artistImage}>
        <a href={props.url} target="_blank">
          <img src={props.image} alt={props.name} />
        </a>
      </div>
      <div className={styles.artistDetails}>
        <span>
          <strong>{props.name}</strong>
        </span>
        {props.detailsInfo.map((item, index) => {
          return <span key={`detailsInfo-${index}`}>{item}</span>;
        })}
      </div>
    </div>
  );
};

export default ArtistCard;

ArtistCard.defaultProps = {
  showClose: false,
  id: "1"
};

ArtistCard.propTypes = {
  /**
   * To show close icon at the top of the component
   */
  showClose: PropTypes.bool,
  /**
   * Callback function after clicking on close icon
   */
  onClose: PropTypes.func,
  /**
   * Name of the artist
   */
  name: PropTypes.string,
  /**
   * ID of the artist
   */
  id: PropTypes.string.isRequired,
  /**
   * Image of the artist
   */
  image: PropTypes.string,
  /**
   * Extra details of an artist
   * It's an array that contains some strings
   */
  detailsInfo: PropTypes.arrayOf(PropTypes.string).isRequired,
  /**
   * Page url of the artist on Brandsintown
   */
  url: PropTypes.string
};
