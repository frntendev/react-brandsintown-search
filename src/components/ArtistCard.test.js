import React from "react";
import { shallow, mount, configure } from "enzyme";
import ArtistCard from "./ArtistCard";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("ArtistCard", () => {
  test("renders component itself correctly", () => {
    const renderedComponent = shallow(
      <ArtistCard
        id="3"
        image="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
        detailsInfo={["name", "family"]}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  test("renders component props correctly", () => {
    const renderedComponent = mount(
      <ArtistCard
        id="3"
        image="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
        detailsInfo={["name", "family"]}
      />
    );
    expect(renderedComponent.props().id).toEqual("3");
    expect(renderedComponent.props().image).toEqual(
      "https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
    );
    expect(renderedComponent.props().detailsInfo).toEqual(["name", "family"]);
  });

  test("Close Icon exists", () => {
    const renderedComponent = mount(
      <ArtistCard
        showClose
        id="3"
        image="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
        detailsInfo={["name", "family"]}
      />
    );
    expect(renderedComponent.find("svg").exists()).toEqual(true);
  });

  test("Close Icon is not exist", () => {
    const renderedComponent = mount(
      <ArtistCard
        id="3"
        image="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
        detailsInfo={["name", "family"]}
      />
    );
    expect(renderedComponent.find("svg").exists()).toEqual(false);
  });
});
