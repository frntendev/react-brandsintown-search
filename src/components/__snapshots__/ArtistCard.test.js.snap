// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCard renders component itself correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistCard
    detailsInfo={
      Array [
        "name",
        "family",
      ]
    }
    id="3"
    image="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
    showClose={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="css-lzlku5-artistImage"
        >
          <a
            target="_blank"
          >
            <img
              src="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
            />
          </a>
        </div>,
        <div
          className="css-1ww3bib-artistDetails"
        >
          <span>
            <strong />
          </span>
          <span>
            name
          </span>
          <span>
            family
          </span>
        </div>,
      ],
      "className": "css-dbjyky-artistRoot",
      "id": "artist-card",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            target="_blank"
          >
            <img
              src="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
            />
          </a>,
          "className": "css-lzlku5-artistImage",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
            />,
            "href": undefined,
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "src": "https://s3.amazonaws.com/bit-photos/large/8518614.jpeg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              <strong />
            </span>,
            Array [
              <span>
                name
              </span>,
              <span>
                family
              </span>,
            ],
          ],
          "className": "css-1ww3bib-artistDetails",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "strong",
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "detailsInfo-0",
            "nodeType": "host",
            "props": Object {
              "children": "name",
            },
            "ref": null,
            "rendered": "name",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "detailsInfo-1",
            "nodeType": "host",
            "props": Object {
              "children": "family",
            },
            "ref": null,
            "rendered": "family",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="css-lzlku5-artistImage"
          >
            <a
              target="_blank"
            >
              <img
                src="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
              />
            </a>
          </div>,
          <div
            className="css-1ww3bib-artistDetails"
          >
            <span>
              <strong />
            </span>
            <span>
              name
            </span>
            <span>
              family
            </span>
          </div>,
        ],
        "className": "css-dbjyky-artistRoot",
        "id": "artist-card",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              target="_blank"
            >
              <img
                src="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
              />
            </a>,
            "className": "css-lzlku5-artistImage",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src="https://s3.amazonaws.com/bit-photos/large/8518614.jpeg"
              />,
              "href": undefined,
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": undefined,
                "src": "https://s3.amazonaws.com/bit-photos/large/8518614.jpeg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                <strong />
              </span>,
              Array [
                <span>
                  name
                </span>,
                <span>
                  family
                </span>,
              ],
            ],
            "className": "css-1ww3bib-artistDetails",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "strong",
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "detailsInfo-0",
              "nodeType": "host",
              "props": Object {
                "children": "name",
              },
              "ref": null,
              "rendered": "name",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "detailsInfo-1",
              "nodeType": "host",
              "props": Object {
                "children": "family",
              },
              "ref": null,
              "rendered": "family",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
